### PR TITLE
New matchers

### PR DESCRIPTION
**Pull request by xxdede**
Added recognition of new brands:

- Sony
- Xiaomi
- Wiko
- LG
- Huawei
- Alcatel
- Vodafone
- Meizu
- Asus

### CODE REVIEW

**Comment by nielsbasjes** [comment]
Thanks! The next few days I'm really busy so I'm probably going to pick this up around the weekend.

File: ua-parser/src/main/java/it/mobimesh/ua/parser/SimpleReader.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                    if (++i % 100 == 0) {
+                        System.out.println(" " + i);
+                    }
+                }
+                
+                System.out.println(" " + i);
+                System.out.println("\nDone!");
+            }
+            
+            bw.flush();
```
Comment (mmorel-35) [inline ua-parser/src/main/java/it/mobimesh/ua/parser/SimpleReader.java]: You should probably flush more often.

File: ua-parser/src/main/java/it/mobimesh/ua/parser/SimpleReader.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        catch(Exception ex) {}        
+    }
+    
+    private static String createSQL(String uaString) {
+        
+        UserAgent ua = USER_AGENT_ANALYZER.parse(uaString);
+        
+        String res = "UPDATE useragent SET ";
+        
+        res += innerSQL("AgentClass",             ua, "Classification",   null);
```
Comment (mmorel-35) [inline ua-parser/src/main/java/it/mobimesh/ua/parser/SimpleReader.java]: Use a StringBuilder?

File: ua-parser/src/main/java/it/mobimesh/ua/parser/SimpleReader.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        System.out.println("\n Done '" + ua.getUserAgentString() + "'.");
+    }
+    
+    private static void readFileAndCreateSQL(String inputPath, String outputPath) {
+        
+        try {
+            
+            BufferedWriter bw = new BufferedWriter(new FileWriter(outputPath));
+            bw.write("");
+
```
Comment (mmorel-35) [inline ua-parser/src/main/java/it/mobimesh/ua/parser/SimpleReader.java]: Why?

File: ua-parser/src/main/java/it/mobimesh/ua/parser/SimpleReader.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        
+        System.out.println("\n Done '" + ua.getUserAgentString() + "'.");
+    }
+    
+    private static void readFileAndCreateSQL(String inputPath, String outputPath) {
+        
+        try {
+            
+            BufferedWriter bw = new BufferedWriter(new FileWriter(outputPath));
+            bw.write("");
```
Comment (mmorel-35) [inline ua-parser/src/main/java/it/mobimesh/ua/parser/SimpleReader.java]: Use try with ressource instead?

File: ua-parser/src/main/java/it/mobimesh/ua/parser/SimpleReader.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        */
+        
+        readFileAndCreateSQL("/tmp/uas.txt", "/tmp/uas.sql");
+    }
+    
+    private static void analyzeAndPrint(String uaString) {
+        
+        System.out.println("Analyzing...\n");
+        UserAgent ua = USER_AGENT_ANALYZER.parse(uaString);
+
```
Comment (mmorel-35) [inline ua-parser/src/main/java/it/mobimesh/ua/parser/SimpleReader.java]: Why not using a real Logger? Log4j2?

**Comment by xxdede** [comment]
@mmorel-35 SimpleReader is a very dummy parsing program that I use to generate sql files and update a MySQL database on which I run queries and verify how the new rules behave.
It did not even have to be part of the commit but I did not know where to put it.

The "real" interesting part of this merge is all in yaml files.

**Comment by mmorel-35** [comment]
I understood , but if you want to merge it please think about it 😊.

**Comment by xxdede** [comment]
Ok, thanks.
I'll do in the future.

Maybe Niels will merge only yaml files.

**Comment by nielsbasjes** [comment]
To make it easier for me can you please add a commit that removes this helper tool you used and any other cleanups you may want to add. 
Thanks.

**Comment by xxdede** [comment]
Sure.

**Pull request closed without merging**